keras/backend/theano_backend.py

```diff
@@ -51,6 +51,9 @@ def placeholder(shape=None, ndim=None, dtype=_FLOATX, name=None):
         return T.tensor3(name=name, dtype=dtype)
     elif ndim == 4:
         return T.tensor4(name=name, dtype=dtype)
+    elif ndim == 5:
+        dtensor5 = T.TensorType(dtype, (False,)*5)
+        return dtensor5(name)
```

Comment:
The latest issue I've seen was knowing whether sample weights would be 2D (sample-wise) or 3D (timestep-wise). Here: https://github.com/fchollet/keras/blob/master/keras/models.py#L470

The current workaround consist in making it possible to specify "temporal" or None as `sample_weight_mode`, which adds quite a bit of unnecessary complexity to the API. Another possible workaround would have been to compile the graph only when `fit` is called (and therefore when we have access to the`sample_weight` data passed by the user) but that would be somewhat hacky (would rely on the user passing consistent weights everytime) and it would require changing the Keras architecture to a large extent.